### PR TITLE
docs: add example for global setup process.env

### PR DESCRIPTION
**Pull request by maheshsundaram**
Add an example for how to attach data to `process.env` in the global setup step to be available in test files. This addresses the confusion I had in #13337

### CODE REVIEW

File: docs/src/test-advanced-js.md [hunk 1]
```diff
@@ -301,6 +301,56 @@ test('test', async ({ page }) => {
 });
 ```
 
+You can make arbitary data available in your tests from your global setup file by attaching it to `process.env`.
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:304]:
```suggestion
You can make arbitrary data available in your tests from your global setup file by setting them as environment variables via `process.env`.
```

File: docs/src/test-advanced-js.md [hunk 2]
```diff
@@ -301,6 +301,56 @@ test('test', async ({ page }) => {
...
+
+```js js-flavor=js
+// global-setup.js
+const { chromium } = require('@playwright/test');
+
+module.exports = async config => {
+  process.env.foo = 'some data';
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:311]: The names of environment variables are case-sensitive. By convention they are usually upper-case. Let's keep them also uppercase in this example?

File: docs/src/test-advanced-js.md [hunk 3]
```diff
@@ -301,6 +301,56 @@ test('test', async ({ page }) => {
...
+Tests have access to the `process.env` properties set in the global setup.
+
+```js js-flavor=ts
+import { test } from '@playwright/test';
+const { foo, bar } = process.env;
+
+test('test', async ({ page }) => {
+  // foo and bar properties are populated.
+  expect(foo).toEqual('some data')
+  expect(bar).toEqual('{"some":"data"}')
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:339]: let's include maybe inside the example `const myComplexData = JSON.parse(process.env.BAR)`? So the parsing is also included.

File: docs/src/test-advanced-js.md [hunk 4]
```diff
@@ -301,6 +301,56 @@ test('test', async ({ page }) => {
 });
 ```
 
+You can make arbitary data available in your tests from your global setup file by attaching it to `process.env`.
+
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:305]: Inside line 214 let's replace 'environment' with 'environment variables' to make it more clear.

**Comment by maheshsundaram** [comment]
@mxschmitt Thank you for the review! I've addressed your feedback with my latest commits. If there is anything else I'd be glad to know.

File: docs/src/test-advanced-js.md [hunk 5]
```diff
@@ -301,6 +301,60 @@ test('test', async ({ page }) => {
 });
 ```
 
+You can make arbitrary data available in your tests from your global setup file by setting them as environment variables via `process.env`.
+
+```js js-flavor=js
+// global-setup.js
+const { chromium } = require('@playwright/test');
+
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:309]:
```suggestion
```

File: docs/src/test-advanced-js.md [hunk 6]
```diff
@@ -301,6 +301,60 @@ test('test', async ({ page }) => {
...
+module.exports = async config => {
+  process.env.FOO = 'some data';
+  // Or a more complicated data structure as JSON:
+  process.env.BAR = JSON.stringify({ some: 'data' });
+};
+```
+
+```js js-flavor=ts
+// global-setup.ts
+import { chromium, FullConfig } from '@playwright/test';
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:319]:
```suggestion
import { FullConfig } from '@playwright/test';
```

File: docs/src/test-advanced-js.md [hunk 7]
```diff
@@ -301,6 +301,60 @@ test('test', async ({ page }) => {
...
+```js js-flavor=ts
+import { test } from '@playwright/test';
+const { FOO, BAR } = process.env;
+
+test('test', async ({ page }) => {
+  // FOO and BAR properties are populated.
+  expect(FOO).toEqual('some data');
+
+  const complexData = JSON.parse(BAR);
+  expect(BAR).toEqual({ some: "data" });
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:341]:
```suggestion
  expect(BAR).toEqual({ some: 'data' });
```

File: docs/src/test-advanced-js.md [hunk 8]
```diff
@@ -301,6 +301,60 @@ test('test', async ({ page }) => {
...
+```js js-flavor=js
+const { test } = require('@playwright/test');
+const { FOO, BAR } = process.env;
+
+test('test', async ({ page }) => {
+  // FOO and BAR properties are populated.
+  expect(FOO).toEqual('some data');
+
+  const complexData = JSON.parse(BAR);
+  expect(BAR).toEqual({ some: "data" });
```
Comment (mxschmitt) [inline docs/src/test-advanced-js.md:354]:
```suggestion
  expect(BAR).toEqual({ some: 'data' });
```

**Review by mxschmitt — APPROVED** [review]

File: docs/src/test-advanced-js.md [hunk 9]
```diff
@@ -301,6 +301,58 @@ test('test', async ({ page }) => {
...
+}
+
+export default globalSetup;
+```
+
+Tests have access to the `process.env` properties set in the global setup.
+
+```js js-flavor=ts
+import { test } from '@playwright/test';
+const { FOO, BAR } = process.env;
```
Comment (aslushnikov) [inline docs/src/test-advanced-js.md:332]: can we inline these? This way it'll be clear that these can be used inside the test, but not outside to declare tests

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**